### PR TITLE
Add a few more python packages

### PR DESCRIPTION
**Pull request by hegner**
Adding
 - py-Genshi
 - py-astroid
 - py-jinja2
 - py-logilab-common
 - py-markupsafe
 - py-mistune
 - py-prettytable
 - py-py2neo
 - py-storm

tested on Ubuntu 16.04 and RedHat 6.7

### CODE REVIEW

**Comment by tgamblin** [comment]
@hegner Thanks!  

Looks like this failed on the new formatting checks.  Mind fixing all but E221?  I should also add one to detect FIXME comments.

**Comment by hegner** [comment]
@tgamblin - sure. However, E203 conflicts with the nice formating in the version definitions. Are you sure you want to keep it?

**Comment by hegner** [comment]
@tgamblin - OK. Did all. However I am hitting a wall with E501 and the long URLs in 
- var/spack/repos/builtin/packages/py-logilab-common/package.py
- var/spack/repos/builtin/packages/py-prettytable/package.py

**Comment by alalazo** [comment]
@hegner For URLs you may try to disable the checks only for that line : 
```
url      = "https://pypi.python.org/packages/a7/31/1650d23e44794d46935d82b86e73454cc83b814cbe1365260ccce8a2f4c6/logilab-common-1.2.0.tar.gz"  # NOQA
```
Use with care though :smile:

**Comment by hegner** [comment]
@alalazo - thanks

**Comment by hegner** [comment]
@tgamblin - I now learned something more about the inner guts of spack. Due to the late binding of the configure commands, one cannot get rid of the F821 problems concerning python, cmake, make, etc

File: var/spack/repos/builtin/packages/py-Genshi/package.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from spack import version, extends, depends_on
+from spack import Package
+
+
+class PyGenshi(Package):
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/py-Genshi/package.py]: can this one be lowercased to `py-genshi` for the package name?  Basically that just requires renaming the directory.

Comment (hegner) [inline var/spack/repos/builtin/packages/py-Genshi/package.py]: done.

**Comment by alalazo** [comment]
@tgamblin @hegner One solution for packages would be to create them with something like (still have to check the exact syntax):
```python
# flake8: ignore=F821
```
What do you think?

**Comment by tgamblin** [comment]
@hegner: I already disabled `E221` for url formatting -- that's in `develop` but this is based on a slightly earlier commit.  

I'll also add `F821` to the list though that might be motivation for me to start putting the things we poke into module scope in the `from spack import *` namespace instead.

**Comment by tgamblin** [comment]
@alalazo: is there a way to add file patterns?  We could just put ignore 821 for `package.py` files in the `flake8.ini` file.  Is there a way to ignore `E501` for particular line patterns?  I like the 79 char limit but I don't want to enforce it for URLs, as @hegner points out.

**Comment by tgamblin** [comment]
@hegner: thanks for being the strict formatting guinea pig

**Comment by hegner** [comment]
@tgamblin - if you can fix it in the __init__ so that E821 can be applied that would be great. I could re-surrect my PR on the spack create in that case. 
E203 vs pretty column alignment is still unclear to me as well.

**Comment by tgamblin** [comment]
@hegner: I actually like `E203`: whitespace before `,`.  I prefer this:

```
    version('2.1',     'abc123')
    version('2.1.5.7', 'abc123')
```

to this:

```
    version('2.1'    , 'abc123')
    version('2.1.5.7', 'abc123')
```

**Comment by tgamblin** [comment]
oops wrong buttong

**Comment by hegner** [comment]
@tgamblin - me as well. could be I took the wrong number in my comment but currently you disallow both.

**Comment by tgamblin** [comment]
@hegner: added F821 t the ignore list... I'll figure out whether I can ignore certain patterns later

**Comment by hegner** [comment]
@tgamblin - OK. I'll fix the outstanding problems for the line lengths.

**Comment by alalazo** [comment]
@tgamblin I'll try to have a look on how to ignore some of the rules just for `packages.py`, let you know if I find something

**Comment by tgamblin** [comment]
@hegner: don't worry about the URL lines -- I can merge it despite those.

**Comment by hegner** [comment]
@tgamblin - OK. In that case it is complete.

**Comment by tgamblin** [comment]
Thanks!

**Pull request merged**